### DISCUSSION TITLE
Noob trying to install GitHub CoPilot

### DISCUSSION THREAD

**Discussion by avin-hailo** [LazyVim/LazyVim#2294] [General]
Hi there

I'm a newcomer to LazyVim and NeoVim.
I've been a Vim user for about years and I'm just now learning lua and the NeoVim / LazyVim configuration style.
I've able to work with it a bit, while working mainly in C++. I have my shortcuts and code completion working fine.

**Now I want to add GitHub CoPilot.**
I started [here](https://www.lazyvim.org/extras) but couldn't understand how to get LazyExtras installed (tried opening the config and typing "extras" - to no avail)
Eventually, I want to get to [here](https://www.lazyvim.org/extras/coding/copilot), but I'm not sure what should I do once I'll get there.

I've used some videos from YouTube and perused the [getting started](https://www.lazyvim.org/) but couldn't understand how to make it happen...

Thanks!

**Comment by dpetka2001** [comment]
To open the `LazyExtras` the easiest way is to restart Neovim and open it again with just `nvim` (don't give any arguments to the invoke command). This will start you on the LazyVim starting page which shows you the `Dashboard`. If you press `x` while on Dashboard, it will open the `LazyExtras` and now you can choose which extra you want to enable. You enable/disable extras by pressing `x` on the ones you want. Or you can also open the UI menu by typing `:LazyExtras` as also mentioned in the docs you linked as well.

**Reply by avin-hailo** [reply]
Thanks!
I think I got it (passed authentication and such) but now I get all the time this warning: _warning: multiple different client offset_encodings detected for buffer, this is not supported yet_ - how to solve this issue?

**Reply by dpetka2001** [reply]
Not 100% sure, but I believe you need [this](https://www.lazyvim.org/configuration/recipes#fix-clangd-offset-encoding).

**Reply by avin-hailo** [reply]
Thanks! Took me a few minutes to figure out where to add it (lzy.lua inside the require("lazy").setup), but now it works. However it seems CoPilot is not that good. I barely get insights from it

**Reply by dpetka2001** [reply]
You'd better put this in a file in `/lua/plugins/` directory as stated in the [docs](https://www.lazyvim.org/configuration#-file-structure) and not pollute the `lazy.lua` file. No idea about copilot, don't use it myself.
